src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1125,21 +1125,19 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
 
   def selectCols(exprs: String*): MatrixTable = {
     val ec = sampleEC
-    val (maybePaths, types, f, isNamedExpr) = Parser.parseSelectExprs(exprs.toArray, ec)
+    val (paths, types, f) = Parser.parseSelectExprs(exprs.toArray, ec)
     val topLevelFields = mutable.Set.empty[String]
-    val finalNames = maybePaths.map(_.toArray).zip(isNamedExpr).map { case (l, named) =>
-      if (named) {
-        assert(l.length == 1)
-        l.head
-      } else {
-        assert(l.head == Annotation.SAMPLE_HEAD)
-        val name = l.last
-        if (l.length == 2) {
-          topLevelFields += name
-        }
-        name
+
+    val finalNames = paths.map {
+      // assignment
+      case Left(name) => name
+      case Right(path) => path match {
+        case List(Annotation.SAMPLE_HEAD, name) => topLevelFields += name
+        case _ =>
```

Comment:
yeah, because that can never happen from Python. We're moving more and more checks out of scala to produce cleaner, shorter code. I can add it back if you want.

Reply:
I think an assert would be nice. it feels like that's what we're doing in other places.

Reply:
I think I also forgot these fixes too. don't re--review yet.